exercise/static/exercise/chapter.js

```diff
@@ -436,18 +417,23 @@
 				//$(form_element).find(":input").prop("disabled", false);
 				//exercise.showLoader("error");
 
-				if (!exercise.active_element) {
+				if (exercise.quiz) {
 					exercise.dom_element.dispatchEvent(
 						new CustomEvent("aplus:exercise-submission-failure",
 							{bubbles: true, detail: {type: exercise.exercise_type}}));
-					exercise.chapter.modalError(exercise.chapter.messages.error);
-				} else {
+					exercise.quizSubmission("error");
+				} else if (exercise.active_element) {
 					// active elements don't use loadbar so the error message must be shown
 					// in the element container
 					var feedback = $("<div>");
 					feedback.attr('id', 'feedback');
 					feedback.append(exercise.chapter.messages.error);
 					exercise.updateOutput(feedback);
+				} else {
+					exercise.dom_element.dispatchEvent(
```

Comment:
Can you look if you can reformat this logic, so that the failure event is only written once and not copied to two if-elif-else branches. E.g. by nesting if statements. This might make the flow of the code easier to follow.

Reply:
Sure! I'll take a look at it later today